Model/Currency.php

```diff
@@ -113,14 +114,18 @@ public function updateRates()
             foreach ($rates as $rate) {
                 $data = $this->getResource()->getByCurrency($rate['Currency']);
                 if (count($data) && isset($data[0]['rate_id'])) {
+                    //update
                     $this->setData([
                         'rate_id'=>$data[0]['rate_id'],
                         'offer_id'=>$rate['Id'],
                         'currency'=>$rate['Currency'],
                         'rate'=>$rate['Rate'],
                         'expire_at'=>$rate['Expiry']
                     ])->save();
+                    $this->storedData = []; //this is one work around for this Magento bug
```

Comment:
just wanted to confirm my understanding, this fixes the Magento bug because the bug was that updates to the database included old information in the storedData array. By clearing this array pre-emptively you're making sure that the next update doesn't have old data getting stored in the database and only valid, up to date information is making it into the database. Is this accurate?

Reply:
Let's say during update scenario (not insert) Magento would like to determine which column of a row has new value. To determine that - previous state of a row should be compared to incoming column values for that exact same row.  Due to a Magento bug (as reported by others on net as well) the comparison was happening between two different rows (with respect to the incoming values for both) instead of previous and new values for the exact same row. 

As a hypothetical example if 
-  In the incoming data two rows have updated value in the `expired_at field  
And
-  The new values in that `expired_at` is same for both rows in the incoming data

Then the `expired_at` was getting updated only for the very first row with this new expired_at value and any rows coming later in the loop with the exact same `expired_at` were incorrectly not getting updated in the database due to that faulty comparison/logic. 

The idea was to clear everything in the storedData associative array (that was holding information about a previous row the code just looked instead of the previous state of the current row). There is a non public function in Magento to do so/reset()/clear which is not available from our extension but is available via $this->storedData (another craziness) . 

 Please let me know if more elaborate explanation is required. It was equally confusing/does not make sense situation for me when I was trying to locate root cause of this problem and had to go several level deep in the call chain to find a logic which seems okay on the surface but it is not due to the way it is comparing one row to another instead of stored value of one row for the incoming /new value for the same row. 

By setting that associative array to an empty one we are forcing Magento  


Due to a Magento bug the code was `comparing` the field values for `two different rows` (related to two different currencies) to decide which fields of current row under consideration have been changed or not. The correct logic would be to compare previous state of the fields (from database) versus incoming values of the fields (from API call) for `the exact same row` (currency).

Reply:
makes more sense now, thank you!